episodes/superspreading-simulate.Rmd

```diff
@@ -0,0 +1,847 @@
+      # creates a column with the chain ID number
+      mutate(chain_id = sim)
+  }
+) %>%
+  # combine list outputs (for each chain ID) into a single data frame
+  list_rbind()
+```
+
+The `sim` element is placed to register the iteration number (**chain ID**) as a new column in the `<epichains>` output. The `purrr::list_rbind()` function aims to combine all the list outputs from `map()`.
+
+:::::::::::::::::::::::::::::::
+
+Now, we are prepared to use `map()` to iterate `simulate_chains()` in a vector from 1 to `r number_chains`:
+
+```{r}
+simulated_chains_map <-
+  # iterate one function across multiple numbers (chain IDs)
+  map(
+    # vector of numbers (chain IDs)
+    .x = seq_len(number_chains),
+    # function to iterate to each chain ID number
+    .f = function(sim) {
+      simulate_chains(
+        # simulation controls
+        index_cases = initial_cases,
+        statistic = "size",
+        # offspring
+        offspring_dist = rnbinom,
+        mu = mers_offspring["mean"],
+        size = mers_offspring["dispersion"],
+        # generation
+        generation_time = function(x) generate(x = serial_inverval, times = x)
+      ) %>%
+        # creates a column with the chain ID number
+        mutate(chain_id = sim)
+    }
+  ) %>%
+  # combine list outputs (for each chain ID) into a single data frame
+  list_rbind()
+```
+
+```{r,echo=FALSE,eval=FALSE}
+# view infectee number per simulation
+simulated_chains_map %>%
+  as_tibble() %>%
+  count(chain_id, sort = TRUE)
+```
+
+::::::::::::::::::::::::::::::::: discussion
+
+### Read the epichains output
+
+```{r,echo=FALSE}
+#### get simulated chain with more infections --------------------------------
+
+chains_subgroup <- simulated_chains_map %>%
+  as_tibble() %>%
+  group_by(chain_id) %>%
+  distinct(infector_id) %>%
+  count(chain_id) %>%
+  filter(n < 5, n >= 3) %>%
+  pull(chain_id)
+
+chain_to_observe <- simulated_chains_map %>%
+  as_tibble() %>%
+  count(chain_id, sort = TRUE) %>%
+  filter(
+    magrittr::is_in(
+      chain_id, chains_subgroup
+    )
+  ) %>%
+  filter(n < 10) %>%
+  filter(n == max(n)) %>%
+  slice_min(chain_id) %>%
+  pull(chain_id)
+```
+
+To explore the output format of the `<epichains>` class object of name `simulated_chains_map`, let's observe closely the simulated `chain_id` number `r chain_to_observe`. 
+
+:::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::: solution
+
+### the epichains object
+
+Let's use `dplyr::filter()` for this:
+
+```{r}
+#### get epichain summary ----------------------------------------------------
+
+simulated_chains_map %>%
+  filter(chain_id == chain_to_observe)
+```
+
+Key elements exclusive from this output are in the **footer**:
+
+```output
+Number of infectors (known): 3
+Number of generations: 3
+```
+
+The simulated `chain_id` number `r chain_to_observe` has three known infectors and three generations. These numbers are more visible when reading the `<epichains>` objects as a data frame.
```

Comment:
See above about `r chain_to_observe` possible typo (will assume any further down in doc are either fine or can be fixed!)